builder/builder.go

```diff
@@ -1737,13 +1741,61 @@ func (b *Builder) BuildImage(format string, template string) error {
 		format = b.State.Mix.Format
 	}
 
-	// If the user did not pass in a template, default to release-image-config.json
+	// swupd (client) called by itser will need a temporary directory to act as its stage dir.
+	wd, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+
+	// If the user did not pass in a template, use the default
 	if template == "" {
-		template = "release-image-config.json"
+		template = imageTemplate
+
+		// If  the default image template is not present in the mix workspace,
+		// copy it from the default ister config directory and update the bundle list based on mix bundles
+		templateFile := filepath.Join(wd, template)
+		if _, err1 := os.Stat(templateFile); os.IsNotExist(err1) {
+			fmt.Printf("Image template %s not found\n", templateFile)
+			var err2 error
+			configFile := filepath.Join(isterConfigDir, template)
+			fmt.Printf("Copying image template %s\n", configFile)
+
+			// Read ister template
+			configValues, err2 := ioutil.ReadFile(configFile)
+			if err2 != nil {
+				return err2
+			}
+			var data map[string]interface{}
+			err2 = json.Unmarshal([]byte(configValues), &data)
+			if err2 != nil {
+				return err2
+			}
+
+			// Read mix bundles
+			bundlesFile := filepath.Join(wd, b.MixBundlesFile)
+			bundles, err2 := helpers.ReadFileAndSplit(bundlesFile) // Note: Returns  trailing empty element if file is empty.
+			// Update bundle list  only if no error and mix bundles not empty
+			if err2 == nil && len(bundles) > 1 {
+				fmt.Printf("Updating image template based on %s\n", bundlesFile)
+				data["Bundles"] = bundles[:len(bundles)-1] // Avoid trailing empty element
```

Comment:
This is a very dangerous way of getting rid of the trailing empty element. If for any reason the mixbundle file does not contain the trailing 0x0a, you will actually remove a valid bundle.  This also doesn't handle empty lines in the middle of the list. I think it is safer to iterate `bundles` and add one element at a time, skipping if the element is empty.

Reply:
@rchiossi Thanks for the comments. But I had mentioned its not ready for review yet. I was in the middle of refactoring the code.

Reply:
ah sorry, I saw it passed travis checks I assumed it was ready for review